### ISSUE TITLE
redis::CannotConnectError not handled

### ISSUE THREAD

**Issue by h0jeZvgoxFepBQ2C** [anycable/action-cable-redis-backport#1]
I tried to play around with this gem and the redis / actioncable disconnects, and it seems that some errors are still not catched by this gem:

redis::CannotConnectError f.e. when you try to connect to redis, but redis is not running.

Since ```redis::CannotConnectError``` is a subclass of ```BaseConnectionError```, but only ```ConnectionError``` is handled, the exception still crashes the server.

I think following line https://github.com/anycable/action-cable-redis-backport/blob/11b7d9ec06371c6837b67821be6835162d459f2d/lib/action-cable-redis-backport.rb#L25 should be replaced from ```ConnectionError``` to ```BaseConnectionError```?

**Comment by h0jeZvgoxFepBQ2C** [comment]
Works fine for our redis version (4.6.0):

```ruby
      if ::Redis::VERSION < "5"
        ConnectionError = ::Redis::BaseConnectionError
      else
        ConnectionError = RedisClient::BaseConnectionError
      end
```

**Comment by palkan** [comment]
Thanks! Fixed in 1.0.1
